core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -178,6 +184,10 @@ private[spark] class HighlyCompressedMapStatus private (
 
   override def location: BlockManagerId = loc
 
+  override def updateLocation(bm: BlockManagerId): Unit = {
```

Comment:
Since location defined inside of each mapstatus we'd have to change that. I've tried to minimize any changes which would impact code paths with this feature turned off.

Reply:
Sounds good.